src/System.Diagnostics.TraceSource/src/System/Diagnostics/CorrelationManager.cs

```diff
@@ -0,0 +1,72 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Threading;
+
+namespace System.Diagnostics
+{
+    public class CorrelationManager
+    {
+        internal CorrelationManager()
+        {
+        }
+
+        private readonly AsyncLocal<Stack> _idStack = new AsyncLocal<Stack>();
```

Comment:
The problem is that it is not the CorrelationManager who is taking care of creating the clone but instead the LogicalCallContext 'knows' to treat a CorrelationManager differently and clone it. 

Since there is no LogicalCallContext, there is no good way that I can see to maintain the same behavior as Desktop. @ericeil any thoughts?